contrib/babelfishpg_tds/src/backend/tds/tdscomm.c

```diff
@@ -893,6 +937,11 @@ TdsReadNextRequest(StringInfo message, uint8_t *status, uint8_t *messageType)
 				TdsDoProcessHeader = true;
 			return 0;
 		}
+		if (TdsRecvMessageType == TDS_BULK_LOAD)
+		{
+			TdsDoProcessHeader = true;
```

Comment:
why are we doing this and at this particular place? Can we do it in TdsReadNextPendingBcpRequest?

Reply:
Because for Bulk Load Request we want to stop reading after the first packet.
Later we read the pending packets using TdsReadNextPendingBcpRequest whenever required

Reply:
but can the logic be driven from TdsReadNextPendingBcpRequest? why we need to do it here only?

Reply:
From GetTdsRequest Function we will call this "TdsReadNextRequest" function. If we do not do it here then we will read the entire BCP Request which we do not want. We instead  we read only the first packet only and then process the headers for that